framework/src/material_image_helpers.cpp

```diff
@@ -2,247 +2,371 @@
-			else {
-				auto path = avk::clean_up_path(mc.mExtraTex);
-				texNamesToUsages[path].push_back(&gm.mExtraTexIndex);
-				if (aLoadTexturesInSrgb) {
-					srgbTextures.insert(path);
+				gm.mExtraTexIndex = -1;
+				if (mc.mExtraTex.empty()) {
+					whiteTexUsages.push_back(&gm.mExtraTexIndex);
 				}
+				else {
+					auto path = avk::clean_up_path(mc.mExtraTex);
+					texNamesToUsages[path].push_back(&gm.mExtraTexIndex);
+					if (aLoadTexturesInSrgb) {
+						srgbTextures.insert(path);
+					}
+				}
+
+				gm.mDiffuseTexOffsetTiling = mc.mDiffuseTexOffsetTiling;
+				gm.mSpecularTexOffsetTiling = mc.mSpecularTexOffsetTiling;
+				gm.mAmbientTexOffsetTiling = mc.mAmbientTexOffsetTiling;
+				gm.mEmissiveTexOffsetTiling = mc.mEmissiveTexOffsetTiling;
+				gm.mHeightTexOffsetTiling = mc.mHeightTexOffsetTiling;
+				gm.mNormalsTexOffsetTiling = mc.mNormalsTexOffsetTiling;
+				gm.mShininessTexOffsetTiling = mc.mShininessTexOffsetTiling;
+				gm.mOpacityTexOffsetTiling = mc.mOpacityTexOffsetTiling;
+				gm.mDisplacementTexOffsetTiling = mc.mDisplacementTexOffsetTiling;
+				gm.mReflectionTexOffsetTiling = mc.mReflectionTexOffsetTiling;
+				gm.mLightmapTexOffsetTiling = mc.mLightmapTexOffsetTiling;
+				gm.mExtraTexOffsetTiling = mc.mExtraTexOffsetTiling;
 			}
-																			 
-			gm.mDiffuseTexOffsetTiling		= mc.mDiffuseTexOffsetTiling	 ;
-			gm.mSpecularTexOffsetTiling		= mc.mSpecularTexOffsetTiling	 ;
-			gm.mAmbientTexOffsetTiling		= mc.mAmbientTexOffsetTiling	 ;
-			gm.mEmissiveTexOffsetTiling		= mc.mEmissiveTexOffsetTiling	 ;
-			gm.mHeightTexOffsetTiling		= mc.mHeightTexOffsetTiling		 ;
-			gm.mNormalsTexOffsetTiling		= mc.mNormalsTexOffsetTiling	 ;
-			gm.mShininessTexOffsetTiling	= mc.mShininessTexOffsetTiling	 ;
-			gm.mOpacityTexOffsetTiling		= mc.mOpacityTexOffsetTiling	 ;
-			gm.mDisplacementTexOffsetTiling	= mc.mDisplacementTexOffsetTiling;
-			gm.mReflectionTexOffsetTiling	= mc.mReflectionTexOffsetTiling	 ;
-			gm.mLightmapTexOffsetTiling		= mc.mLightmapTexOffsetTiling	 ;
-			gm.mExtraTexOffsetTiling		= mc.mExtraTexOffsetTiling		 ;
 		}
 
 		std::vector<avk::image_sampler> imageSamplers;
-		const auto numSamplers = texNamesToUsages.size() + (whiteTexUsages.empty() ? 0 : 1) + (straightUpNormalTexUsages.empty() ? 0 : 1);
+
+		size_t numTexUsages = texNamesToUsages.size();
+		size_t numWhiteTexUsages = whiteTexUsages.empty() ? 0 : 1;
+		size_t numStraightUpNormalTexUsages = (straightUpNormalTexUsages.empty() ? 0 : 1);
+
+		if (aSerializer) {
+			aSerializer->get().archive(numTexUsages);
+			aSerializer->get().archive(numWhiteTexUsages);
+			aSerializer->get().archive(numStraightUpNormalTexUsages);
+		}
+
+		const auto numSamplers = numTexUsages + numWhiteTexUsages + numStraightUpNormalTexUsages;
 		imageSamplers.reserve(numSamplers);
 
-		auto getSync = [numSamplers, &aSyncHandler, lSyncCount = size_t{0}] () mutable -> avk::sync {
+		auto getSync = [numSamplers, &aSyncHandler, lSyncCount = size_t{ 0 }]() mutable->avk::sync {
```

Comment:
The many formatting changes are a bit annoying. You don't have to revert them, but please check your Visual Studio's settings and disable ~"format on paste" or similar. (Might also be in ReSharper's settings if you are using that).